include/pybind11/pytypes.h

```diff
@@ -1638,6 +1638,13 @@ class capsule : public object {
     }
 
     const char *name() const { return PyCapsule_GetName(m_ptr); }
+
+    /// Replaces a capsule's name *without* calling the destructor on the existing one.
+    void set_name(const char *&new_name) {
```

Comment:
Why `const char *&new_name` instead of just `const char *`?
```suggestion
    void set_name(const char *new_name) {
```

Also: 
"If non-NULL, the name must outlive the capsule. If the previous name stored in the capsule was not NULL, no attempt is made to free it."

Shouldn't we try to free the old name? In our `set_name` function?